specs/interop.md

```diff
@@ -0,0 +1,708 @@
+included in the source chain. There is no strict requirement that the execution message is ever submitted.
+
+The term "block builder" is used interchangeably with the term "sequencer" for the purposes of this document but
+they need not be the same entity in practice.
+
+## The Dependency Set
+
+The dependency set defines the set of chains that a destination chains allows as source chains. Another way of
+saying it is that the dependency set defines the set of initiating messages that are valid for an executing
+message to be included. An executing message MUST have an initiating message that is included in a chain
+in the dependency set.
+
+The dependency set is defined by a set of chain ids. Since it is impossible to enforce uniqueness of chain ids,
+social consensus MUST be used to determine the chain that represents the canonical chain id. This
+particularly impacts the block builder as they SHOULD use the chain id to assist in validation
+of executing messages.
+
+The dependency set is configured on a per chain basis.
+
+The chainid of the local chain MUST be considered as part of its own dependency set.
+
+While the dependency set explicitly defines the set of chains that are depended on for incoming messages,
+the full set of transitive dependencies must be known to allow for the progression of [safety](#safety).
+
+### Chain ID
+
+The concept of a chain id was introduced in [EIP-155](https://eips.ethereum.org/EIPS/eip-155) to prevent
+replay attacks between chains. This EIP does not specify the max size of a chain id, although
+[EIP-2294](https://eips.ethereum.org/EIPS/eip-2294) attempts to add a maximum size. Since this EIP is
+stagnant, all representations of chain ids MUST be the `uint256` type.
+
+In the future, OP Stack chains reserve the right to use up to 32 bytes to represent a chain id. The
+configuration of the chain should deterministically map to a chain id and with careful architecture
+changes, all possible OP Stack chains in the superchain will be able to exist counterfactually.
+
+It is a known issue that not all software in the Ethereum space can handle 32 byte chain ids.
+
+## Interop Network Upgrade
+
+The interop network upgrade timestamp defines the timestamp at which all functionality in this document is considered
+the consensus rules for an OP Stack based network. On the interop network upgrade block, a set of deposit transaction
+based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
+
+- L1 Attributes Transaction calling `setL1BlockValuesEcotone`
+- User deposits from L1
+- Network Upgrade Transactions
+  - L1Block deployment
+  - CrossL2Inbox deployment
+  - L2ToL2CrossDomainMessenger deployment
+  - Update L1Block Proxy ERC-1967 Implementation Slot
+  - Update CrossL2Inbox Proxy ERC-1967 Implementation Slot
+  - Update L2ToL2CrossDomainMessenger Proxy ERC-1967 Implementation Slot
+  - Mint Cross Chain Ether Liquidity in L2ToL2CrossDomainMessenger
```

Comment:
Is this a new deposit transaction type, similar to the [L1 Attributes](https://specs.optimism.io/protocol/deposits.html#l1-attributes-deposited-transaction) deposit transaction, that sets the `type(uint248).max` initial balance?